test/lib/smart-survey-api.test.js

```diff
@@ -2,106 +2,60 @@
-
-      context = {
-        createClient: sinon.stub(),
-        client: {
-          getResponses: sinon.stub()
-        }
-      };
-      const callback = sinon.stub();
-
-      // Causes the stub to call the argument at the index as a callback function.
-      // stub.callsArg(2); stub calls the 3rd argument as a callback.
-      // callsArgWith like callsArg, but with arguments to pass to the callback.
-      context.client.getResponses.callsArgWith(2, 'error message', 1);
-      smartSurveyClientAPI.getData.call(context, 'a', 'b', undefined, callback);
-      callback.should.have.been.calledWith('error message', 1);
-    });
-    it('should call the callback without an error when we receive success from the api', function() {
-
-      context = {
-        createClient: sinon.stub(),
-        client: {
-          getResponses: sinon.stub()
-        }
-      };
-      const callback = sinon.stub();
-
-      context.client.getResponses.callsArgWith(2, null, 'good');
-      smartSurveyClientAPI.getData.call(context, 'a', 'b', undefined, callback);
-      callback.should.have.been.calledWith(null, 'good');
-    });
+// 2nd argument: An object with the key: module name to proxy & the value the object above
+// So this means smartSurveyStub will now point to this function, i.e. a stub
+const smartSurveyAPI = proxyquire('../../lib/smart-survey-api', { 'smartsurvey-client': smartSurveyStub});
+
+describe('smartSurveyAPI', () => {
+  describe('getData()', () => {
+    it('is a function', () => (typeof smartSurveyAPI.getData).should.equal('function'));
+    // it('is takes 3 arguments', () => (typeof smartSurveyAPI.getData).should.have.lengthOf(3));
+    describe('when HAPPYPATH', () => {
+      let result;
+      const response = {foo: 'bar'};
+
+      before(() => {
+        getResponsesStub
+          .withArgs('survey1', sinon.match.any)
+          .yields(null, response);
+        result = smartSurveyAPI.getData('mytoken', 'mytokensecret', 'survey1');
+      });
+
+      after(() => {
+        getResponsesStub.reset();
+      });
+
+      it('returns a Promise', () => result.should.be.a('Promise'));
+      it('resolves', () => result.should.eventually.be.fulfilled);
+      it('that resolves to the response', () => result.should.eventually.equal(response));
   });
-
-  describe('parseResponse', function() {
-    it('should return an error message if there is an error', function() {
-      parseResponse('error', null).should.equal('error');
+    describe('when NOT HAPPYPATH', () => {
+      let result;
+      const error = 'some error';
+      before(() => {
+        getResponsesStub
+          .withArgs('noSurveyId', sinon.match.any)
+          .yields(error, null);
+        result = smartSurveyAPI.getData('mytoken', 'mytokensecret', 'noSurveyId');
+      });
+      after(() => {
+        getResponsesStub.reset();
+      });
+      it('returns a Promise', () => result.should.be.a('Promise'));
```

Comment:
ok so I think I get it now.  Every promise that is rejected should be dealt with (usually with a catch).  However, since chai-as-promised has rejectedWith(blah) that explains that this rejection was dealt with in the argument.

So it has been dealt with in line 53 below